google/resource_google_project.go

```diff
@@ -396,18 +399,19 @@ func resourceGoogleProjectDelete(d *schema.ResourceData, meta interface{}) error
 func resourceProjectImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	parts := strings.Split(d.Id(), "/")
 	pid := parts[len(parts)-1]
-	// Prevent importing via project number, this will cause issues later
 	matched, err := regexp.MatchString("^\\d+$", pid)
 	if err != nil {
 		return nil, fmt.Errorf("Error matching project %q: %s", pid, err)
 	}
-
 	if matched {
-		return nil, fmt.Errorf("Error importing project %q, please use project_id", pid)
+		// this is actually a project number not a project id
+		d.Set("number", pid)
+	} else {
+		d.Set("project_id", pid)
 	}
-
 	// Ensure the id format includes projects/
-	d.SetId(fmt.Sprintf("projects/%s", pid))
```

Comment:
Can you add a comment here and at the top of Read() saying that project ID might be either project ID or number?